### ISSUE TITLE
Issue when overriding an Interface

### ISSUE THREAD

**Issue by FeernandoOFF** [ForteScarlet/kotlin-suspend-transform-compiler-plugin#67]
Hello,First I'd like to thank your project it has been a life saver and the last piece that I needed to start my Kotlin-JS interop project so, thanks!

I'm experiencing an issue where I have an Interface that has a `suspend` function in it and an Implementation but it seems that the `override` goes to the exported JS, so Given the following code the compiler will throw the next exception:

```
'getDataAsync' overrides nothing.
```

Interface:
```kotlin
interface FooInterface {
    suspend fun getData(): String
}
```
Implementation:
```kotlin
@JsExport
class Foo(
    private val httpClient: HttpClient = HttpModule().client
): FooInterface  {
    @JsExport.Ignore
    @JsPromise
    override suspend fun getData(): String {
        val response = httpClient.get(CivRoutes()).bodyAsText()
        return response
    }
}
```
Is there a way that the generated JS just ignores the `override`?

**Comment by ForteScarlet** [comment]
Hello! I'm glad this plugin helped you! 😊

Currently it looks like the `override` of the synthesized function will come directly from the source function causing this problem.
The fastest temporary solution is to add the appropriate annotations directly to the interface's pending functions as well:
```kotlin
interface FooInterface {
    @JsPromise
    suspend fun getData(): String
}
```

I'll try to fix this in a later release, like making the synthesized function also look to the parents to see if an inheritable function exists, or... Maybe just marking `open` will fix it?

**Comment by FeernandoOFF** [comment]
Update: It didn't 😆 

Generated with the workaround:
```
export declare class Foo /* implements FooInterface */ {
    constructor(httpClient?: any/* HttpClient */);
}
export declare namespace Foo {
    class $getDataCOROUTINE$0 /* extends CoroutineImpl */ {
        constructor(_this_: Foo, resultContinuation: any/* Continuation<string> */);
    }
}
```

**Comment by FeernandoOFF** [comment]
Thanks! <3

**Comment by ForteScarlet** [comment]
👌
It will be released in the next *-0.9.3 release.
Maybe it will be 2.1.0-0.9.3? if kt's 2.1.0 is coming soon XD
